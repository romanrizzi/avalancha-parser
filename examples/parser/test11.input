check
-- Precedencia
not
not
x1
and
not
not
x2
or
not
not
x3
and
not
not
x4
imp
not
not
x5
and
not
not
x6
or
not
not
x7
and
not
not
x8
