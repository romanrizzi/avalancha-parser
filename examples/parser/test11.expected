[
   "program",
   [],
   [
      [
         "check",
         [
            "imp",
            [
               "or",
               [
                  "and",
                  [
                     "not",
                     [
                        "not",
                        [
                           "equal",
                           [
                              "var",
                              "x1"
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "not",
                        [
                           "equal",
                           [
                              "var",
                              "x2"
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ]
                     ]
                  ]
               ],
               [
                  "and",
                  [
                     "not",
                     [
                        "not",
                        [
                           "equal",
                           [
                              "var",
                              "x3"
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "not",
                        [
                           "equal",
                           [
                              "var",
                              "x4"
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ]
                     ]
                  ]
               ]
            ],
            [
               "or",
               [
                  "and",
                  [
                     "not",
                     [
                        "not",
                        [
                           "equal",
                           [
                              "var",
                              "x5"
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "not",
                        [
                           "equal",
                           [
                              "var",
                              "x6"
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ]
                     ]
                  ]
               ],
               [
                  "and",
                  [
                     "not",
                     [
                        "not",
                        [
                           "equal",
                           [
                              "var",
                              "x7"
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "not",
                        [
                           "equal",
                           [
                              "var",
                              "x8"
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ]
   ]
]
